composer.json

```diff
@@ -44,7 +44,7 @@
         "firebase/php-jwt": "^6.3",
         "google/protobuf": "3.22.3",
         "grpc/grpc": "1.52.0",
-        "psr/log": "^1.1 || ^2.0 || ^3.0",
```

Comment:
if we leave this in, i won't be able to user newer versions of the sdk. i need psr log v3.

Reply:
if you want to get a php 7.4 compatible lock file, you can use the platform config to make composer pretend it's a particular version of php. https://getcomposer.org/doc/06-config.md#platform. for example: https://github.com/GrahamCampbell/GitHub-Notifications/blob/4.1/composer.json#L51.

Reply:
@GrahamCampbell Understood, this isn't intended to be merged. I was trying to see if there was even a theoretical way for me to have a `composer.lock` acceptable to both 7 and 8, and it looks like it is not.

The conditional class definition worked fine and tests are passing with all versions now. We'll be migrating that change into another PR in process, and I'm just doing experiments on this branch.

Reply:
setting the platform to 7.4.15 would work on php 7.4 through 8.3, but it would not let you run the tests on the newer libs.

Reply:
@GrahamCampbell is the change in 438eab9 what you were suggesting? If so, it's producing another mole to whack as it's causing the PHP 8 code in the conditional class definition I just added to [fail for PHP 8 again](https://github.com/momentohq/client-sdk-php/actions/runs/7836168717/job/21383177523?pr=178) (as it's pulling in psr/log 1.1.4). If you had something different in mind, could you please elaborate?

Reply:
The conditional class generation is incorrect. Both cases should never have param types, and only the PHP case should have the return types. PHP users can install any of the 3 versions of the package, so the PHP version doesn't tell you what lib version. If v1 is installed, having param types is illegal.

Reply:
@GrahamCampbell I see, thanks. I don't have a tremendous understanding of the implications of that platform setting apart from, as you mentioned, it's going to favor older library versions meaning we won't be testing more recent versions in CI. I definitely want to be certain that this can't cause breaking changes for existing users of the SDK. This config change and the new composer.lock don't run the risk of downgrading packages that were installed previously, do they?

Reply:
no. the platform config and composer.lock file are not inherited or visible at all to package consumers. they are only relevant to people directly working on this repo.

Reply:
Perfect, thank you.